lib/modules/datasource/rubygems/get.ts

```diff
@@ -182,4 +207,113 @@ export class InternalRubyGemsDatasource extends Datasource {
+    key: (nexusEndPoint: string) => `isNexusDataSource:${nexusEndPoint}`,
+    ttlMinutes: 1440,
+  })
+  async isNexusDataSource(nexusEndPoint: string): Promise<boolean> {
+    const statusEndPoint = '/service/rest/v1/status';
+    const nexusUrl = nexusEndPoint.concat(statusEndPoint);
+    let isNexus = false;
+    try {
+      const response = await this.http.getJson<HttpResponse>(nexusUrl);
+      isNexus = response.headers?.server?.includes('Nexus') ?? false;
+    } catch (err) {
+      return false;
+    }
+    return isNexus;
+  }
+
+  private async getReleasesFromNexus(
+    repository: string,
+    nexusRegistry: string,
+    dependencyName: string
+  ): Promise<Release[]> {
+    const result: Release[] = [];
+    const gemItems: NexusGemsItems[] = [];
+    const url = this.getNexusGemReleasesEndPoint(
+      repository,
+      nexusRegistry,
+      dependencyName
+    );
+    try {
+      const nexusResponse = (await this.http.getJson<NexusGems>(url))?.body || {
+        continuationToken: null,
+        items: [],
+      };
+      gemItems.push(...nexusResponse.items);
+      const continuationToken: string | null = nexusResponse.continuationToken;
+      if (continuationToken) {
+        const paginationGems = await this.paginationResult(
+          url,
+          continuationToken
+        );
+        gemItems.push(...paginationGems);
+      }
+    } catch (error) {
+      logger.debug('Failed to retreive gem from nexus');
+      return [];
+    }
+
+    result.push(
+      ...gemItems.map((item) => {
+        return { version: item.version, rubyPlatform: 'nexus' };
+      })
+    );
+    return result;
+  }
+
+  private getNexusGemReleasesEndPoint(
+    repository: string,
+    nexusRegistry: string,
+    depName: string
+  ): string {
+    const nexusSearchPoint = '/service/rest/v1/search';
+    const query = `${nexusSearchPoint}?repository=${repository}&name=${depName}&sort=version`;
+    const gemReleasesEndPoint = nexusRegistry.concat(query);
+    return gemReleasesEndPoint;
+  }
+
+  private async paginationResult(
+    url: string,
+    continuationToken: string
+  ): Promise<NexusGemsItems[]> {
+    const gemItems: NexusGemsItems[] = [];
+    let pageToken: string | null = continuationToken;
+    while (pageToken) {
+      const gemsOfPage: NexusGems = await this.pageOfNexusGems(url, pageToken);
+      gemItems.push(...gemsOfPage.items.filter(Boolean));
+      pageToken = gemsOfPage.continuationToken;
+    }
+    return gemItems;
+  }
+
+  private async pageOfNexusGems(
+    url: string,
+    pageToken: string
+  ): Promise<NexusGems> {
+    let nexusPage: NexusGems = { continuationToken: null, items: [] };
+    try {
+      const paginateUrl = url.concat(`&continuationToken=${pageToken}`);
+      const response = await this.http.getJson<NexusGems>(paginateUrl);
+      nexusPage = response.body;
+    } catch (error) {
+      logger.debug('Failed to retreive gems page from nexus');
+      throw error;
+    }
```

Comment:
```suggestion
      const paginateUrl = url.concat(`&continuationToken=${pageToken}`);
      const response = await this.http.getJson<NexusGems>(paginateUrl);
      return response.body;
```

Don't catch and rethrow. this makes no benefit. We should discard whole result instead and let upper layer catch the error.